packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2305,13 +2306,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // export default interface allowed in:
       // https://github.com/Microsoft/TypeScript/pull/16040
       if (this.state.value === "interface") {
-        const result = this.tsParseDeclaration(
-          this.startNode(),
-          this.state.value,
-          true,
-        );
-
-        if (result) return result;
+        if (this.lookahead().type === tt.name) {
+          const result = this.tsParseDeclaration(
+            this.startNode(),
+            this.state.value,
+            true,
+          );
+          if (result) return result;
+        } else {
+          this.next();
+          this.raise(this.state.pos, TSErrors.IdentifierExpected);
```

Comment:
If we do what I wrote above we have already skipped interface, so we can 
```suggestion
          this.raise(this.state.start, TSErrors.IdentifierExpected);
```